lib/cm3/nvic.c

```diff
@@ -161,6 +163,23 @@ void nvic_set_priority(uint8_t irqn, uint8_t priority)
 		/* Device specific interrupts */
 		NVIC_IPR(irqn) = priority;
 	}
+#else
+        volatile uint32_t *block;
+        unsigned int index;
+	if (irqn < NVIC_IRQ_COUNT) {
```

Comment:
is this relying on it being unsigned and that the negatives will appear to be higher?

Reply:
Same code as original code